reframe/core/shell.py

```diff
@@ -98,8 +98,10 @@ def write_body(self, s):
         self.write(s, 'body')
 
     def finalize(self):
-        ret = '\n'.join([self.shebang, *self._prolog,
-                         *self._body, *self._epilog])
+        # Here empty strings are filtered out before joining
```

Comment:
Write it as "Filter out empty statements before returning". And return directly, without using the `stripped_shell` variable.

Reply:
Nonetheless, have you found the place that actually generated these empty instructions? Perhaps, we could fix the problem there.

Reply:
I would also expand the shell script generator tests to check this stripping.

Reply:
As discussed offline, I think it's better to fix that in the Slurm job script generator.